pkg/controller/nodenetworkstateconfiguration/nodenetworkstateconfiguration_controller.go

```diff
@@ -139,9 +142,50 @@ func (r *ReconcileNodeNetworkStateConfiguration) Reconcile(request reconcile.Req
 
 	nmstateOutput, err := nmstate.ApplyDesiredState(instance)
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("error reconciling nodenetworkstate configuration at desired state apply: %v", err)
+		errmsg := fmt.Errorf("error reconciling nodenetworkstate at desired state apply: %v", err)
+		conditions.SetCondition(
+			instance,
+			nmstatev1alpha1.NodeNetworkStateConditionFailing,
+			corev1.ConditionTrue,
+			"Failed",
+			errmsg.Error(),
+		)
+		conditions.SetCondition(
+			instance,
+			nmstatev1alpha1.NodeNetworkStateConditionAvailable,
+			corev1.ConditionFalse,
+			"Success",
+			"successfully reconciled NodeNetworkState",
+		)
+		r.client.Status().Update(context.TODO(), instance)
+		return reconcile.Result{}, errmsg
 	}
 	reqLogger.Info("nmstate", "output", nmstateOutput)
 
+	for i := 0; i < 5; i++ {
```

Comment:
We have two controllers writting into the status at the same time - the status reporter and this setup. Therefore simultaneous writes to the field can happen (quite often since it takes a lot of time to handle the reconcile and get here to write the status). Therefore we try several times to handle the collision.